File: 28. Find the Index of the First Occurrence in a String.java
class Solution {
    public int strStr(String haystack, String needle) {
        int size = needle.length();
        for(int index = 0; index < haystack.length()-size+1;index++){
            if(needle.equals(haystack.substring(index,index+size))){
                return index;
            }
        }
        return -1;
    }
}

File: 28. Find the Index of the First Occurrence in a String.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        nLength = len(needle)
        if(len(haystack) == nLength):
            if(haystack == needle): return 0
            return -1
        for x in range(1+len(haystack)-nLength):
            if(haystack[x:x+nLength]==needle): return x
        return -1
        